### DISCUSSION TITLE
Standard ML on Zulip

### DISCUSSION THREAD

**Discussion by dmacqueen** [smlnj/smlnj#266] [Announcements]
There is now an organization on Zulip, standardml.zulipchat.com, with chat "streams" for asking questions or commenting on Standard ML, the language, and SML/NJ, the system (compiler, runtime, libraries, tools).

Use the link: https://standardml.zulipchat.com/join/3oyu7lpkjo2tqskwe3mhz7am/
to subscribe to the Zulip Standard ML organization.

**Comment by dmacqueen** [comment]
The Zulip Standard ML organization is now up to 30 members! It includes two of the authors of the Definition of Standard ML, Revised, and at least a couple experienced SML compiler developers.

**Comment by spdegabrielle** [comment]
How do you get an invite to join?
the link has expired:

<img width="710" alt="image" src="https://github.com/smlnj/smlnj/assets/108047/dc4329f7-8d45-480b-a5b0-e357fdcabf2d">

**Reply by dmacqueen** [reply]
> How do you get an invite to join? the link has expired:
> 
> <img alt="image" width="710" src="https://user-images.githubusercontent.com/108047/237123901-dc4329f7-8d45-480b-a5b0-e357fdcabf2d.png">

Here is a new link to join the Standard ML organization at Zulip.com

https://standardml.zulipchat.com/join/dlxbftoozdxy3dfa7b6ucp6b/

**Reply by dmacqueen** [reply]
> How do you get an invite to join? the link has expired:
> 
> <img alt="image" width="710" src="https://user-images.githubusercontent.com/108047/237123901-dc4329f7-8d45-480b-a5b0-e357fdcabf2d.png">

I try to create invitation links that have no expiration, but I may have slipped up on this one.  Or maybe non-expiring invitation links in fact expire sometimes?

**Comment by spdegabrielle** [comment]
Thank you

How would someone new to Standard ML find out about the zulip chat? 

e.g. visiting  http://www.smlnj.org/ or https://smlfamily.github.io/ (via https://en.wikipedia.org/wiki/Standard_ML)

**Reply by dmacqueen** [reply]
Very good and important question.  My plan was to start advertising the Zulip Standard ML organization once it became established (and after I had definitely decided on Zulip vs Discord as a chat service).  We should put a link (a non-expiring invitation link) on smlnj.org and the GitHub project and probably smlfamily.org.

**Comment by gasche** [comment]
I am also unable to join the Zulip chat due because it is invite-only. My impression is that if you want to make it convenient for people to join the Zulip, it would be better if no invitation was required to join, as is standard in other community zulips (for example <https://typ.zulipchat.com/>).

**Comment by dmacqueen** [comment]
That is a good suggestion.  When I set up the zulip Standard ML organization, I was unaware of the fact that it could be configured to allow people to join without an invitation.  I’ll figure out how to change the configuration to make that possible.

Thanks,

Dave MacQueen


> On Oct 25, 2025, at 5:19 AM, Gabriel Scherer ***@***.***> wrote:
> 
> 
> I am also unable to join the Zulip chat due because it is invite-only. My impression is that if you want to make it convenient for people to join the Zulip, it would be better if no invitation was required to join, as is standard in other community zulips (for example https://typ.zulipchat.com/).
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/orgs/smlnj/discussions/266#discussioncomment-14779706>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAGXNPOWYBLRIPEKY7OXP533ZNTDPAVCNFSM6AAAAACKGCEZISVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTINZXHE3TANQ>.
> You are receiving this because you authored the thread.
> 

David MacQueen
***@***.***

**Comment by gasche** [comment]
I believe that the relevant configuration option is in "Settings icon > Organization settings > Organization permissions", then un-check "Invitations are required for joining this organization".

**Comment by dmacqueen** [comment]
Thanks.  I have unchecked the requirement for invitations to join the Zulip organization.

Dave


> On Oct 26, 2025, at 12:37 PM, Gabriel Scherer ***@***.***> wrote:
> 
> 
> I believe that the relevant configuration option is in "Settings icon > Organization settings > Organization permissions", then un-check "Invitations are required for joining this organization".
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/orgs/smlnj/discussions/266#discussioncomment-14786676>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAGXNPM6ZKAL7ZVIQDNCUJ33ZUIEZAVCNFSM6AAAAACKGCEZISVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTINZYGY3DONQ>.
> You are receiving this because you authored the thread.
> 

David MacQueen
***@***.***
